# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsPIDOMWindow.h
# Commit: 0e3bc029e59f
# Full Hash: 0e3bc029e59f5f0319a57c469d7c0991081379ff
# Author: Benjamin VanderSloot <bvandersloot@mozilla.com>
# Date: 2025-06-10 21:43:01
# Regressor Bug: 1969017
# File Overlap Count: 2
# Description:
#   Bug 1969017 - Refactor FedCM to use a managed actor rather than tagging onto WindowGlobalParent - r=anti-tracking-reviewers,emz
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D251617
# ==============================================================================

diff -r 8b2ffdf05455 -r 0e3bc029e59f dom/base/nsPIDOMWindow.h
--- a/dom/base/nsPIDOMWindow.h	Tue Jun 10 11:49:57 2025 +0000
+++ b/dom/base/nsPIDOMWindow.h	Tue Jun 10 11:49:57 2025 +0000
@@ -66,6 +66,7 @@
 class ServiceWorkerDescriptor;
 class Timeout;
 class TimeoutManager;
+class WebIdentityHandler;
 class WindowContext;
 class WindowGlobalChild;
 class CustomElementRegistry;
@@ -621,6 +622,8 @@
 
   bool UsingStorageAccess();
 
+  mozilla::dom::WebIdentityHandler* GetOrCreateWebIdentityHandler();
+
   uint32_t UpdateLockCount(bool aIncrement) {
     MOZ_ASSERT_IF(!aIncrement, mLockCount > 0);
     mLockCount += aIncrement ? 1 : -1;
@@ -666,6 +669,7 @@
 
   RefPtr<mozilla::dom::Performance> mPerformance;
   mozilla::UniquePtr<mozilla::dom::TimeoutManager> mTimeoutManager;
+  RefPtr<mozilla::dom::WebIdentityHandler> mWebIdentityHandler;
 
   RefPtr<mozilla::dom::Navigation> mNavigation;
 